### PR TITLE
Terraform example docs

### PR DESCRIPTION
**Pull request by th31nitiate**
Fixes issue #483 

This add the Terraform example documentation as described in the issue. I attempted to explained things the way I understood. 

I also made sure to perform spell/grammer checking to avoid wasting your time while performing the review.

## Proposed Changes

  - Add Terraform usage example documentation

### CODE REVIEW

File: docs/example-terraform.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# Terraform example
+
+We will be looking at how to use Kapitan to compile terraform files with Jsonnet as the input type. It's possible to use other input types, however, Jsonnet is recommended. 
+For example, we could use the Kadet input to generate terraform files but this would require templates to be written in YAML then to rendered into JSON.
```
Comment (uberspot) [inline docs/example-terraform.md:4]: "then rendered into JSON" *

File: docs/example-terraform.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+Jsonnet is the most suitable input type as you will see due to its functional nature. The only appropriate output type is JSON since this is the format that Terraform consumes.
```
Comment (uberspot) [inline docs/example-terraform.md:5]: Maybe i'd rephrase "Jsonnet is the most suitable input type as you will see due to its functional nature." to "Jsonnet is the most straightforward input type as you will see due to its functional nature.". Suitable sounds a bit as if Kapitan is biased between input types. :)

File: docs/example-terraform.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+export TF_DATA_DIR=$(realpath -m ${DIR}/../../../.TF_DATA_DIR/{{inventory.parameters.name}}) # Folder for TF initialization (preferable outside of compiled)
+export OUTPUT_DIR=$(realpath -m ${DIR}/../../../output/{{inventory.parameters.name}}) # Folder for storing output files (preferable outside of compiled)
+```
+
+It is good practice to utilize this method to improve integration with various CLI based tools. Scripts help to ensure terraform &
+kapitan can function with your CI/CD systems. It generally depends on your organizational workflows. 
+
+### secrets
+
+Although there are no particular secrets in this instance. It is possible to utilize Kapitan secrets as defined in [secrets management](secrets.md). There is truly no limit to the extensibility of this tool.
```
Comment (uberspot) [inline docs/example-terraform.md:187]: "There is truly no limit to the extensibility of this tool." - Definitely appreciate the comment :) but I think it might be misinterpreted if we add it in the terraform docs? 
Might be better to keep this doc purely about terraform?

File: docs/example-terraform.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+When terraform runs it will use the output module to generate your desired `README.md` using information from terraform's state.
+
+Scripts are located in the `scripts` directory. They are compiled using jinja2 as the templating language. An example is as follows:
+
+```shell
+export TF_DATA_DIR=$(realpath -m ${DIR}/../../../.TF_DATA_DIR/{{inventory.parameters.name}}) # Folder for TF initialization (preferable outside of compiled)
+export OUTPUT_DIR=$(realpath -m ${DIR}/../../../output/{{inventory.parameters.name}}) # Folder for storing output files (preferable outside of compiled)
+```
+
+It is good practice to utilize this method to improve integration with various CLI based tools. Scripts help to ensure terraform &
```
Comment (uberspot) [inline docs/example-terraform.md:182]: All this looks great. :) Appreciate the effort!
One last comment is if you could replace `&` with `and` as it makes searching codebases/documentation more intuitive?

File: docs/example-terraform.md (hunk 1 above)
Comment (ramaro) [reply docs/example-terraform.md:4]: This is actually not true. Kadet will work with both JSON or YAML.

Comment (th31nitiate) [reply docs/example-terraform.md:4]: Thanks for spotting that. 

@ramaro Thanks that sounds important. It means integration would be much simpler since we can convert HCL to JSON. Kadet can then use the pre-existing JSON as  input templates.

File: docs/example-terraform.md (hunk 2 above)
Comment (th31nitiate) [reply docs/example-terraform.md:5]: That is true. It is important to highlight that Kapitan is not biased on input types. It is dynamic which enables creativity.

File: docs/example-terraform.md (hunk 3 above)
Comment (th31nitiate) [reply docs/example-terraform.md:187]: I felt like I was not doing the secrets section much justice.

I do agree with you on that one though.

File: docs/example-terraform.md (hunk 4 above)
Comment (th31nitiate) [reply docs/example-terraform.md:182]: I never considered that before. Good point :D
